CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+
+cmake_minimum_required(VERSION 3.14)
+# Set project name, version and laguages here. (change as needed)
+# Version numbers are available by including "exampleConfig.h" in
+# the source. See exampleConfig.h.in for some more details.
+project(TRAJECTORY_PLANNING VERSION 1.0.0 LANGUAGES CXX)
+
+find_package( Boost COMPONENTS system filesystem iostreams REQUIRED ) #whatever libs you need
+include_directories( ${Boost_INCLUDE_DIRS} ${PROJECT_SOURCE_DIR}/include)
+
+# Options: Things you can set via commandline options to cmake (e.g. -DENABLE_LTO=[ON|OFF])
+option(ENABLE_WARNINGS_SETTINGS "Allow target_set_warnings to add flags and defines.
+                                 Set this to OFF if you want to provide your own warning parameters." ON)
+option(ENABLE_LTO "Enable link time optimization" ON)
+option(ENABLE_GOOGLETESTS "Include tests in the library. Setting this to OFF will remove all googletest related code.
+Tests in tests/*.cpp will still be enabled." ON)
+
+# Include stuff. No change needed.
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
+include(ConfigSafeGuards)
+include(Colors)
+include(CTest)
+include(Googletest)
+include(Documentation)
+include(LTO)
+include(Misc)
+include(Warnings)
+
+# Check for LTO support.
+find_lto(CXX)
+
+# --------------------------------------------------------------------------------
+#                         Locate files (change as needed).
+# --------------------------------------------------------------------------------
+set(SOURCES          # All .cpp files in src/
+    src/trajectory.cpp
+)
+set(TESTFILES        # All .cpp files in tests/
+    tests/tests.cpp
+)
+set(LIBRARY_NAME engine)  # Default name for the library built from src/*.cpp (change if you wish)
+
+# --------------------------------------------------------------------------------
+#                            Build! (Change as needed)
+# --------------------------------------------------------------------------------
+# Compile all sources into a library.
+add_library(${LIBRARY_NAME} OBJECT ${SOURCES})
+
+# Lib needs its header files, and users of the library must also see these (PUBLIC). (No change needed)
+target_include_directories(${LIBRARY_NAME} PUBLIC ${PROJECT_SOURCE_DIR}/include)
+
+# Set the compile options you want (change as needed).
+target_set_warnings(${LIBRARY_NAME} ENABLE ALL AS_ERROR ALL DISABLE Annoying)
+# target_compile_options(${LIBRARY_NAME} ... )  # For setting manually.
+
+# Add an executable for the file app/main.cpp.
+# If you add more executables, copy these lines accordingly.
+add_executable(main app/main.cpp)   # Name of exec. and location of file.
+target_link_libraries(main PRIVATE ${LIBRARY_NAME} ${Boost_LIBRARIES})  # Link the executable to library (if it uses it).
+target_set_warnings(main ENABLE ALL AS_ERROR ALL DISABLE Annoying) # Set warnings (if needed).
+target_enable_lto(main optimized)  # enable link-time-optimization if available for non-debug configurations
```

Comment:
Is this a new feature in cmake? 
I was expecting compile flags like `-Wall -Wpedantic -O2` but these lines seem to do similar work. Correct me otherwise.

Reply:
No, it is making use of `cmake/Warnings.cmake`